src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp

```diff
@@ -314,141 +314,105 @@ inline double less_or_equal(double a, double b) {
-        std::nth_element(abs_values.begin(), abs_values.begin() + (expected_shape - 1) / 2, abs_values.end());
-        abs_median = (abs_values[(expected_shape - 1) / 2] + abs_values[expected_shape / 2]) / 2.0;
+    void get_results() {
+        if (!incorrect_values.empty()) {
+            std::string msg = "[ COMPARATION ] COMPARATION IS FAILED! incorrect elem counter: ";
+            msg += std::to_string(incorrect_values.size());
+            msg += ". Please print `incorrect_values` to get detailed information!";
+            throw std::runtime_error(msg);
+        }
     }
-    return abs_median;
-}
+};
 
 template <typename ExpectedT, typename ActualT>
-void compare(const ov::Tensor& expected,
-             const ov::Tensor& actual,
-             const double abs_threshold_ = std::numeric_limits<double>::max(),
-             const double rel_threshold_ = std::numeric_limits<double>::max()) {
+void compare(const ov::Tensor& expected, const ov::Tensor& actual, double abs_threshold, const double rel_threshold) {
     auto expected_shape = expected.get_shape();
     auto actual_shape = actual.get_shape();
     if (expected_shape != actual_shape) {
         std::ostringstream out_stream;
         out_stream << "Expected and actual shape are different: " << expected_shape << " " << actual_shape;
         throw std::runtime_error(out_stream.str());
-    }
-    if (shape_size(actual_shape) == 0) {
+    } else if (shape_size(actual_shape) == 0) {
         return;
     }
 
-    auto expected_data = expected.data<ExpectedT>();
-    auto actual_data = actual.data<ActualT>();
-    double abs_threshold = abs_threshold_;
-    double rel_threshold = rel_threshold_;
-    size_t shape_size_cnt = shape_size(expected_shape);
-    if (abs_threshold == std::numeric_limits<double>::max() && rel_threshold == std::numeric_limits<double>::max()) {
-        if (sizeof(ExpectedT) == 1 || sizeof(ActualT) == 1) {
-            abs_threshold = 1.;
-        } else {
-            std::vector<double> abs_values(shape_size_cnt);
-            for (size_t i = 0; i < shape_size_cnt; i++) {
-                abs_values[i] = std::fabs(static_cast<double>(expected_data[i]));
-            }
-            auto abs_median = calculate_median(abs_values);
-            abs_threshold = abs_median * 0.05 < 1e-5 ? 1e-5 : 0.05 * abs_median;
-
-            if (std::is_integral<ExpectedT>::value) {
-                abs_threshold = std::ceil(abs_threshold);
-            }
-        }
-    }
-    if (!std::isnan(abs_threshold)) {
-        std::cout << "[ COMPARATION ] rel_threshold: " << rel_threshold << std::endl;
-    }
-    if (!std::isnan(rel_threshold)) {
-        std::cout << "[ COMPARATION ] abs_threshold: " << abs_threshold << std::endl;
+    const auto default_abs_threshold = get_eps_by_ov_type(expected.get_element_type());
+    if (abs_threshold == std::numeric_limits<double>::max()) {
+        abs_threshold = default_abs_threshold;
```

Comment:
For the moment `abs_threshold` has no default value. And it's risky to compare floats by using `==`, so I'd suggest to use negative value (e.g. `-1`) in order to check was custom threshold provided, or need to use default value

Reply:
default values are set to std::numeric_limits<double>::max()

Reply:
The default values will be changed to -1 in separate PR